benchmarks/perfetto/s3_utils.py

```diff
@@ -0,0 +1,35 @@
+import os
+from pathlib import Path
+from . import BENCHMARK_ROOT
+
+
+def decompress_s3_data(s3_tarball_path: Path):
+    assert str(s3_tarball_path.absolute()).endswith(
+        ".tar.gz"
+    ), f"Expected .tar.gz file path but get {s3_tarball_path}."
+    import tarfile
+
+    # Hide decompressed file in .data directory so that they won't be checked in
+    decompress_dir = os.path.join(BENCHMARK_ROOT, ".data", s3_tarball_path.name.rstrip(".tar.gz"))
+    os.makedirs(decompress_dir, exist_ok=True)
+    print(f"decompressing input tarball: {s3_tarball_path}...", end="", flush=True)
+    tar = tarfile.open(s3_tarball_path)
+    tar.extractall(path=decompress_dir)
+    tar.close()
+    print("OK")
+
+
+def checkout_s3_data(name: str, decompress: bool = True):
+    S3_URL_BASE = "https://ossci-datasets.s3.amazonaws.com/torchbench"
+    download_dir = Path(BENCHMARK_ROOT).joinpath(".data")
+    download_dir.mkdir(parents=True, exist_ok=True)
+    import requests
+
+    full_path = download_dir.joinpath(name)
+    s3_url = f"{S3_URL_BASE}/traces/{name}"
+    r = requests.get(s3_url, allow_redirects=True)
```

Comment:
We should use `raise_for_status` to check for an HTTP error, right?

Reply:
I guess it is fine, we don't have special handling for failed HTTP error, users have to re-run the command